src/viam/app/location/client.py

```diff
@@ -0,0 +1,720 @@
+    NewRobotPartResponse,
+    NewRobotRequest,
+    NewRobotResponse,
+    Robot,
+)
+from viam.proto.app import RobotPart as RobotPartPB
+from viam.proto.app import (
+    SharedSecret,
+    ShareLocationRequest,
+    ShareLocationResponse,
+    UnshareLocationRequest,
+    UnshareLocationResponse,
+    UpdateLocationRequest,
+    UpdateLocationResponse,
+    UpdateRobotPartRequest,
+    UpdateRobotPartResponse,
+    UpdateRobotRequest,
+    UpdateRobotResponse,
+)
+from viam.utils import dict_to_struct, struct_to_dict
+
+LOGGER = logging.getLogger(__name__)
+
+
+class LocationClient:
+    """gRPC client for method calls to app.
+
+    Constructor is used by `AppClient` to instantiate relevant service stub. Calls to `LocationClient`  methods should be made through
+    `AppClient`.
+    """
+
+    class RobotPart:
+        """`RobotPart` shadow class.
+
+        Copy the attributes of a `viam.proto.app.RobotPart` object and convert and store its internal `Timestamp` objects as Python
+        `datetime` objects. It also converts the relevant `Struct` objects into dictionaries, mapping strings to items of type `Any`.
+        """
+
+        @classmethod
+        def robot_part_from_proto(cls, robot_part: RobotPartPB) -> Self:
+            """Create a `RobotPart` from the .proto defined `RobotPart`.
+
+            Args:
+                robot_part (viam.proto.app.RobotPart): The object to copy from.
+
+            Returns:
+                RobotPart: The `RobotPart`.
+            """
+            self = cls()
+            self.proto = robot_part
+            self.id = robot_part.id
+            self.name = robot_part.name
+            self.dns_name = robot_part.dns_name
+            self.secret = robot_part.secret
+            self.robot = robot_part.robot
+            self.location_id = robot_part.location_id
+            self.robot_config = struct_to_dict(robot_part.robot_config) if robot_part.HasField("robot_config") else None
+            self.last_access = robot_part.last_access.ToDatetime() if robot_part.HasField("last_access") else None
+            self.user_supplied_info = struct_to_dict(robot_part.user_supplied_info) if robot_part.HasField("user_supplied_info") else None
+            self.main_part = robot_part.main_part
+            self.fqdn = robot_part.fqdn
+            self.local_fqdn = robot_part.fqdn
+            self.created_on = robot_part.created_on.ToDatetime() if robot_part.HasField("created_on") else None
+            self.secrets = robot_part.secrets
+            return self
+
+        id: str
+        name: str
+        dns_name: str
+        secret: str
+        robot: str
+        location_id: str
+        robot_config: Optional[Mapping[str, Any]]
+        last_access: Optional[datetime]
+        user_supplied_info: Optional[Mapping[str, Any]]
+        main_part: bool
+        fqdn: str
+        local_fqdn: str
+        created_on: Optional[datetime]
+        secrets: List[SharedSecret]
+
+        proto: RobotPartPB
```

Comment:
Make this a property that generates a new `RobotPartPB` every time, like

```python3
@property
def proto(self) -> RobotPartPB:
    return RobotPartPB(
        id = self.id,
        name = self.name,
        ...
    )
```

That way, if they change any of the attributes on the native `RobotPart`, it'll be reflected in the new `RobotPartPB`.